src/SFML/Window/Win32/WindowImplWin32.cpp

```diff
@@ -886,12 +886,15 @@ void WindowImplWin32::processEvent(UINT message, WPARAM wParam, LPARAM lParam)
             if (m_keyRepeatEnabled || ((HIWORD(lParam) & KF_REPEAT) == 0))
             {
                 Event::KeyPressed event;
-                event.alt      = HIWORD(GetKeyState(VK_MENU)) != 0;
-                event.control  = HIWORD(GetKeyState(VK_CONTROL)) != 0;
-                event.shift    = HIWORD(GetKeyState(VK_SHIFT)) != 0;
-                event.system   = HIWORD(GetKeyState(VK_LWIN)) || HIWORD(GetKeyState(VK_RWIN));
-                event.code     = virtualKeyCodeToSF(wParam, lParam);
-                event.scancode = toScancode(wParam, lParam);
+                event.alt        = HIWORD(GetKeyState(VK_MENU)) != 0;
+                event.control    = HIWORD(GetKeyState(VK_CONTROL)) != 0;
+                event.shift      = HIWORD(GetKeyState(VK_SHIFT)) != 0;
+                event.system     = HIWORD(GetKeyState(VK_LWIN)) || HIWORD(GetKeyState(VK_RWIN));
+                event.capsLock   = HIWORD(GetKeyState(VK_CAPITAL)) != 0;
+                event.numLock    = HIWORD(GetKeyState(VK_NUMLOCK)) != 0;
+                event.scrollLock = HIWORD(GetKeyState(VK_SCROLL)) != 0;
```

Comment:
```suggestion
                event.capsLock   = (LOWORD(GetKeyState(VK_CAPITAL)) & 1) != 0;
                event.numLock    = (LOWORD(GetKeyState(VK_NUMLOCK)) & 1) != 0;
                event.scrollLock = (LOWORD(GetKeyState(VK_SCROLL)) & 1) != 0;
```

You need LOWORD instead of HIWORD to get the toggled state instead of the current pressed state (see [GetKeyState function return value](https://learn.microsoft.com/en-us/windows/win32/api/winuser/nf-winuser-getkeystate#return-value)).

When pressing one of the lock keys, the LOWORD values were `65408` or `65409` instead of just `0` or `1`, which is why I've added the `& 1`.

Note that when pressing the lock key itself, Windows will give you the NEW state while Linux will give you the OLD state. I don't think this matters much in practice though, the important thing is that all key presses afterwards have the correct state.

Reply:
That's what I get for copy pasting and not reading the docs properly 🙃 

Thank you!

Reply:
I might need to update the documentation for macOS.  
macOS doesn't really know "Num Lock" it's not something they support.

I'll try to test it on our Mac Mini.

Reply:
> That's what I get for copy pasting and not reading the docs properly 🙃

Talking about reading the docs, I think the code is still completely wrong (even though it works).
I used LOWORD because the existing code was using HIWORD, but the usage of HIWORD already seems wrong.

HIWORD and LOWORD are used to get the 16-bit words from a 32-bit DWORD, but `GetKeyState` only returns a 16-bit short to begin with. My first instinct was that HIWORD should always have returned 0, but apparently the returned 16-bit value is negative and the implicit conversion from WORD to DWORD in the HIWORD call causing all high-bits to become 1 (which caused to existing code to actually work).

When pressing e.g. the F1 key, the result of `GetKeyState(VK_F1)` is always `-127` or `-128` (i.e. `10000001` or `10000000` in binary). The highest bit (which happens to be the sign bit) is 1 in both these values (because the key is being pressed). The lowest bit alternates between each key press as the state is toggled.

The same two values (`-127` and `-128`) of course also appear from `GetKeyState(VK_CAPITAL)` when pressing the caps lock key itself. When calling `GetKeyState(VK_CAPITAL)` while pressing another key though, we instead get the value `0` or `1`, as now the caps lock key isn't being held down and only the lowest bit changes depending on whether the caps lock is on or off.

So the code should probably look like this:
```c++
event.shift      = (GetKeyState(VK_SHIFT) & 0x8000) != 0;  // Checking if key is currently down
event.scrollLock = (GetKeyState(VK_CAPITAL) & 0x0001) != 0;  // Checking if lock state is on
```

Reply:
You're right!

I've also removed the `!= 0` check, since it doesn't really do anything.